applications/sentence-transformers-ft/main.py

```diff
@@ -0,0 +1,97 @@
+from datasets import load_dataset
+from torch.utils.data import DataLoader
+from sentence_transformers import SentenceTransformer, InputExample, losses, evaluation
+import modal
+import pathlib
+from datetime import datetime
+
+model_id = "BAAI/bge-small-en-v1.5"
+dataset_id = "quora"
+
+# Modal constants
+VOL_MOUNT_PATH = pathlib.Path("/vol")
+GPU_CONFIG = "a10g"
+
+
+# Modal resources
+volume = modal.Volume.persisted(
+    f"sentence-transformers-ft-{int(datetime.now().timestamp())}"
+)
+stub = modal.Stub("finetune-embeddings")
+image = modal.Image.debian_slim().pip_install(
+    "sentence-transformers", "torch", "datasets"
+)
+
+# Quora pairs dataset: https://huggingface.co/datasets/quora
+dataset = load_dataset(dataset_id, split="train")
+# Quora pairs dataset only contains a "train" split in huggingface, so we will manually split it into train and test
+train_test_split = dataset.train_test_split(test_size=0.1, seed=42)
+train_dataset = train_test_split["train"]
+test_dataset = train_test_split["test"]
+
+
+@stub.function(
+    image=image,
+    gpu=GPU_CONFIG,
+    timeout=15000,
+    volumes={VOL_MOUNT_PATH: volume},
+    _allow_background_volume_commits=True,
+)
+def finetune():
+    """
+    Finetune a sentence transformer on the quora pairs dataset. Evaluates model performance before/after training
+
+    Inspired by: https://github.com/UKPLab/sentence-transformers/blob/657da5fe23fe36058cbd9657aec6c7688260dd1f/examples/training/quora_duplicate_questions/training_MultipleNegativesRankingLoss.py
+    """
+    model = SentenceTransformer(model_id)
+
+    train_examples = []
+    for i in range(train_dataset.num_rows):
```

Comment:
use for item in list

rather than 

```python
for i in range(len(list)):
    list[i]
```

this could mostly be done with a list comprehension like 


```
test_examples = [InputExample(texts=[item["questions"]["text"][0], item["questions"]["text"][1]], label=int(item["is_duplicate"])) for item in test_dataset]
evaluator = evaluation.BinaryClassificationEvaluator.from_input_examples(test_examples)
```

Reply:
I coded it that way because I was temporarily training on a fraction of the dataset ( I.e train_dataset.num_rows // 100), but yea I'll change it to be more pythonic